### PR TITLE
Use a better error when a cljs repl form cannot be found

### PR DESCRIPTION
**Pull request by arichiardi**
Without this fix cider throws an error but it is not a very clear one.

-----------------

Before submitting the PR make sure the following things have been done (and denote this
by checking the relevant checkboxes):

- [X] The commits are consistent with our [contribution guidelines][1]
- [X] All tests are passing (`make test`)
- [X] All code passes the linter (`make lint`) which is based on [`elisp-lint`](https://github.com/gonewest818/elisp-lint) and includes
  - This fails with exit = 2 but I don't think it due to my change.

- [X] You've updated the [changelog][3] (if adding/changing user-visible functionality)

### CODE REVIEW

**Comment by bbatsov** [comment]
Hmm, I'm not sure that suppressing the error is a good idea. Probably it should just be made more meaningful using `user-error`. After your changes simply nothing's going to happen if people mistyped/misconfigured something, which is going to be pretty confusing.

**Comment by arichiardi** [comment]
Uhm True, will try a better fix

File: cider.el [hunk 1]
```diff
@@ -755,7 +755,7 @@ you're working on."
 
 (defun cider-cljs-repl-form (repl-type)
   "Get the cljs REPL form for REPL-TYPE."
-  (let ((repl-form (cadr (seq-find
+  (when-let ((repl-form (cadr (seq-find
```
Comment (bbatsov) [inline cider.el]: I guess you can just use `if-let` here and show an `user-error` in the `else` branch. Something like `No ClojureScript REPL type x found. Please check ...`

**Comment by arichiardi** [comment]
@bbatsov  done ;)

File: cider.el [hunk 2]
```diff
@@ -755,14 +755,15 @@ you're working on."
...
-      repl-form)))
+  (if-let ((repl-form (cadr (seq-find
+                             (lambda (entry)
+                               (eq (car entry) repl-type))
+                             cider-cljs-repl-types))))
+      ;; repl-form can be either a string or a function producing a string
+      (if (symbolp repl-form)
+          (funcall repl-form)
+        repl-form)
+    (user-error "No ClojureScript REPL type x found.  Please make sure that `cider-cljs-repl-types` has an entry for `cljs-repl-type`")))
```
[the remark this answers is not in the document]
Comment (arichiardi) [reply cider.el]: Ah got you! Let me google Elisp interpolations :D

[the remark this answers is not in the document]
Comment (arichiardi) [reply cider.el]: oh I thought Elisp had a construct for interpolations :D

File: cider.el [hunk 3]
```diff
@@ -755,14 +755,15 @@ you're working on."
...
   "Get the cljs REPL form for REPL-TYPE."
-  (let ((repl-form (cadr (seq-find
-                          (lambda (entry)
-                            (eq (car entry) repl-type))
-                          cider-cljs-repl-types))))
-    ;; repl-form can be either a string or a function producing a string
-    (if (symbolp repl-form)
-        (funcall repl-form)
-      repl-form)))
+  (if-let ((repl-form (cadr (seq-find
```
Comment (bbatsov) [inline cider.el]: I see this in the build log `cider.el:756:30:Error: `(repl-form (cadr (seq-find (lambda (entry) (eq (car entry) repl-type)) cider-cljs-repl-types)))' is a malformed function`. If think by mistake someone made `if-let` work with just one set of parens and the form was immediately deprecated afterwards in Emacs 26.1. See how we're using this elsewhere in the code.

File: cider.el [hunk 4]
```diff
@@ -755,14 +755,15 @@ you're working on."
...
+                             (lambda (entry)
+                               (eq (car entry) repl-type))
+                             cider-cljs-repl-types))))
+      ;; repl-form can be either a string or a function producing a string
+      (if (symbolp repl-form)
+          (funcall repl-form)
+        repl-form)
+    (user-error (format "No ClojureScript REPL type %s found.  Please make sure that `cider-cljs-repl-types` has an entry for it" repl-type))))
```
Comment (bbatsov) [inline cider.el]: I think `user-error` does `format` implicitly inside. You can check the docstring.

**Comment by bbatsov** [comment]
👍

**Pull request merged**